atc/worker/pool.go

```diff
@@ -231,19 +231,17 @@ func (pool Pool) findWorkerForContainer(logger lager.Logger, owner db.ContainerO
 		return nil, nil, false, err
 	}
 
+	for _, w := range workersWithContainer {
```

Comment:
@clarafu I don't think this is a problem. In theory, it should never find more than 1 container, and when the found container was created, it should have checked team comparability.

Reply:
I think it becomes a problem for users that don't have global-resources enabled. Since they won't be sharing the same version history, it does matter where the check is run because they will all have individual scopes and will need to be checked individually. We also documented the feature that if they use team workers, all work/build will be scheduled onto the team workers.

Reply:
I believe Concourse has goal of enabling global-resources by default, and I don't see a reason of not enabling global-resources, thus I believe disabling global-resources is minority. So, if this is the only concern, we can have this optimization only when global-resources is enabled.

Reply:
I really want it to be on by default 😢  But we still have the problem of https://github.com/concourse/concourse/issues/3023 that we need to fix before we can turn it on by default